### PR TITLE
Keep OverlapDetector generic but normalize the getters for BedRecord and Interval

### PR DESCRIPTION
**Pull request by nh13**
Alternate to #34

### CODE REVIEW

File: pybedlite/bed_record.py [hunk 1]
```diff
@@ -183,6 +182,26 @@ def bed_fields(self) -> List[str]:
             ),
         ]
 
+    @property
+    def reference_name(self) -> str:
```
Comment (nh13) [inline pybedlite/bed_record.py:186]: I could be convinced this should just be called `refname` to match `Interval`

**Review by clintval — COMMENTED** [review]
I like it but I would only ask we add property names to make the interface between `BedRecord` and `Interval` consistent and not add the `zero_based_start` and `zero_based_open_end` at this time.

File: pybedlite/bed_record.py (hunk 1 above)
Comment (clintval) [reply pybedlite/bed_record.py:186]: Yes!

File: pybedlite/overlap_detector.py [hunk 2]
```diff
@@ -177,85 +180,35 @@ def from_bedrecord(cls: Type["Interval"], record: BedRecord) -> "Interval":
         )
 
 
-_GenericGenomicSpan = TypeVar(
-    "_GenericGenomicSpan",
-    bound=Union[_GenomicSpanWithChrom, _GenomicSpanWithContig, _GenomicSpanWithRefName],
-)
+GenericGenomicsSpan = TypeVar("GenericGenomicsSpan", bound=Union[GenomicSpan, StrandedGenomicSpan])
```
Comment (msto) [inline pybedlite/overlap_detector.py:183]:
```suggestion
GenericGenomicSpan = TypeVar("GenericGenomicSpan", bound=Union[GenomicSpan, StrandedGenomicSpan]) 
```

This should be named consistently with the protocols.

File: pybedlite/overlap_detector.py [hunk 3]
```diff
@@ -177,85 +180,35 @@ def from_bedrecord(cls: Type["Interval"], record: BedRecord) -> "Interval":
...
 """
-A generic genomic feature where the reference sequence name is accessed with any of the 3 most
-common property names ("chrom", "contig", "refname"). This type variable is used for describing the
+A generic genomic feature. This type variable is used for describing the
 generic type contained within the :class:`~pybedlite.overlap_detector.OverlapDetector`.
 """
 
 
-class OverlapDetector(Generic[_GenericGenomicSpan], Iterable[_GenericGenomicSpan]):
+class OverlapDetector(Generic[GenericGenomicsSpan], Iterable[GenericGenomicsSpan]):
```
Comment (msto) [inline pybedlite/overlap_detector.py:190]: Why is `GenericGenomicSpan` necessary? `StrandedGenomicSpan` subclasses `GenomicSpan`, why not just use `GenomicSpan` here?

**Review by msto — CHANGES_REQUESTED** [review]

**Review by clintval — COMMENTED** [review]
What do you think about adding a `negative` property to `BedRecord` so we have fewer duplicated getters across both classes instead of introducing `is_negative` to both?

File: pybedlite/overlap_detector.py [hunk 4]
```diff
@@ -137,6 +135,11 @@ def __attrs_post_init__(self) -> None:
         if self.end <= self.start:
             raise ValueError(f"end <= start: {self.end} <= {self.start}")
 
+    @property
+    def is_negative(self) -> bool:
+        """True if the interval is on the negative strand, False otherwise"""
+        return self.negative
```
[the remark this answers is not in the document]
Comment (nh13) [reply pybedlite/overlap_detector.py:141]: bug

File: pybedlite/bed_record.py [hunk 5]
```diff
@@ -183,6 +182,16 @@ def bed_fields(self) -> List[str]:
...
 
+    @property
+    def refname(self) -> str:
+        """A reference sequence name."""
+        return self.chrom
+
+    @property
+    def is_negative(self) -> bool:
+        """True if the interval is on the negative strand, False otherwise"""
+        return self.strand is not None and self.strand == BedStrand.Positive
```
[the remark this answers is not in the document]
Comment (clintval) [reply pybedlite/bed_record.py:193]: @nh13 bug here

**Review by clintval — APPROVED** [review]

**Pull request merged**